### PR TITLE
Fix-93-Custom SSL Changes for WLS admin only domain setup

### PR DESCRIPTION
**Pull request by gnsuryan**

### CODE REVIEW

File: src/main/arm/nestedtemplates/adminTemplate.json [hunk 1]
```diff
@@ -356,7 +363,7 @@
                "fileUris": [
                   "[uri(parameters('_artifactsLocationAdminTemplate'), concat('../scripts/', variables('name_scriptFile'), parameters('_artifactsLocationSasToken')))]"
                ],
-               "commandToExecute": "[concat('sh',' ',variables('name_scriptFile'),' ',parameters('wlsDomainName'),' ',parameters('wlsUserName'),' ',parameters('wlsPassword'),' ',parameters('adminVMName'),' ', variables('const_wlsHome'),' ')]"
+               "commandToExecute": "[concat('sh',' ',variables('name_scriptFile'),' ',parameters('wlsDomainName'),' ',parameters('wlsUserName'),' ',parameters('wlsPassword'),' ',parameters('adminVMName'),' ', variables('const_wlsHome'),' ',string(parameters('enableHTTPAdminListenPort')),' ')]"
```
Comment (edburns) [inline src/main/arm/nestedtemplates/adminTemplate.json]: See https://github.com/wls-eng/arm-oraclelinux-wls/issues/158 Move `commandToExecute` properties that include passwords to `protectedSettings`

File: src/main/arm/nestedtemplates/adminTemplateForCustomSSL.json [hunk 2]
```diff
@@ -0,0 +1,557 @@
...
+                    ],
+                    "commandToExecute": "[concat('sh',' ',variables('name_scriptFile'),' ',parameters('wlsDomainName'),' ',parameters('wlsUserName'),' ',parameters('wlsPassword'),' ',parameters('adminVMName'),' ', variables('const_wlsHome'),' ',string(parameters('enableHTTPAdminListenPort')),' ',string(parameters('enableCustomSSL')),' ',if(parameters('enableCustomSSL'),base64(parameters('wlsSSLCustomIdentityKeyStoreData')),''),' ',if(parameters('enableCustomSSL'),base64(parameters('wlsSSLCustomIdentityKeyStorePassPhrase')),''),' ',if(parameters('enableCustomSSL'),base64(parameters('wlsSSLCustomIdentityKeyStoreType')),''),' ',if(parameters('enableCustomSSL'),base64(parameters('wlsSSLCustomTrustStoreData')),''),' ',if(parameters('enableCustomSSL'),base64(parameters('wlsSSLCustomTrustStorePassPhrase')),''),' ',if(parameters('enableCustomSSL'),base64(parameters('wlsSSLCustomTrustStoreType')),''),' ',if(parameters('enableCustomSSL'),base64(parameters('wlsSSLServerPrivateKeyAlias')),''),' ',if(parameters('enableCustomSSL'),base64(parameters('wlsSSLServerPrivateKeyPassPhrase')),''),' ')]"
+                }
```
Comment (edburns) [inline src/main/arm/nestedtemplates/adminTemplateForCustomSSL.json]: See https://github.com/wls-eng/arm-oraclelinux-wls/issues/158 Move `commandToExecute` properties that include passwords to `protectedSettings`

**Review by edburns — CHANGES_REQUESTED** [review]
Hello @gnsuryan , I am certainly not opposed to accepting this merge, but I have one requested change and one question.

* Requested change: See https://github.com/wls-eng/arm-oraclelinux-wls/issues/158 Move `commandToExecute` properties that include passwords to `protectedSettings` 

* Comment: what about the changes to `createUiDefinition.json`?  Issue https://github.com/wls-eng/arm-oraclelinux-wls/issues/93 calls for this to be optional, so I think the enable/disable, as well as providing the keyvault data, needs to be available in the UI, right?

File: src/main/arm/nestedtemplates/adminTemplateForCustomSSL.json (hunk 2 above)
Comment (gnsuryan) [reply src/main/arm/nestedtemplates/adminTemplateForCustomSSL.json]: Thanks Ed. I will move the commandToExecute properties under protectedSettings. 

Regarding changes to createUIDefinition.json, as you have mentioned, we would require additional fields that will accept user inputs regarding keyvault resource group and keyvault data. 

The template though works without them as well, as I have made them optional and also defaulted the customSSLEnabled flag to false.

I will retest the changes for protected settings and create a new pull request. You may want to reject the existing merge request.

**Comment by wls-eng** [comment]
Closing/cancelling this pull request as I found that there are more changes that need to be done to support Custom SSL certificate implementation for WebLogic Admin Console.

**Pull request closed without merging**